File: src/main/java/com/github/kindrat/cassandra/client/ui/keylistener/TableCellCopyHandler.java
package com.github.kindrat.cassandra.client.ui.keylistener;

import com.github.kindrat.cassandra.client.ui.DataObject;
import javafx.collections.ObservableList;
import javafx.scene.control.TablePosition;
import javafx.scene.control.TableView;
import javafx.scene.input.Clipboard;
import javafx.scene.input.ClipboardContent;
import lombok.RequiredArgsConstructor;

import java.util.Objects;

@RequiredArgsConstructor
public class TableCellCopyHandler implements Runnable {
    private final TableView<DataObject> tableView;

    @Override
    public void run() {
        Clipboard clipboard = Clipboard.getSystemClipboard();
        ClipboardContent content = new ClipboardContent();

        int row = tableView.getSelectionModel().getSelectedIndex();
        ObservableList<TablePosition> selectedCells = tableView.getSelectionModel().getSelectedCells();
        if (!selectedCells.isEmpty()) {
            TablePosition tablePosition = selectedCells.get(0);
            Object cellData = tablePosition.getTableColumn().getCellData(row);
            content.putString(Objects.toString(cellData));
            clipboard.setContent(content);
        }
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/keylistener/TableCellCopyHandler.kt
package com.github.kindrat.cassandra.client.ui.keylistener

import com.github.kindrat.cassandra.client.ui.DataObject
import com.github.kindrat.cassandra.client.util.getCellData
import javafx.scene.control.TableView
import javafx.scene.input.Clipboard
import javafx.scene.input.ClipboardContent
import java.util.*

class TableCellCopyHandler(private val tableView: TableView<DataObject>) : Runnable {

    override fun run() {
        val clipboard = Clipboard.getSystemClipboard()
        val content = ClipboardContent()
        val row = tableView.selectionModel.selectedIndex
        val selectedCells = tableView.selectionModel.selectedCells
        if (!selectedCells.isEmpty()) {
            val tablePosition = selectedCells[0]
            val cellData: Any? = tablePosition.tableColumn.getCellData(row, row)
            content.putString(Objects.toString(cellData))
            clipboard.setContent(content)
        }
    }
}